src/measure_theory/measure/measure_space.lean

```diff
@@ -2863,6 +2863,12 @@ protected lemma is_finite_measure_on_compacts.smul [topological_space α] (μ :
   is_finite_measure_on_compacts (c • μ) :=
 ⟨λ K hK, ennreal.mul_lt_top hc (hK.measure_lt_top).ne⟩
 
+@[priority 50] -- see Note [lower instance priority]
+instance compact_space.is_finite_measure
+  [topological_space α] [compact_space α] [is_finite_measure_on_compacts μ] :
+  is_finite_measure μ :=
```

Comment:
It seems that we have an instance in the other direction: a finite measure is finite on compacts. You can add it as a lemma but not as an instance (at least, before we migrate to Lean 4).

Reply:
Thanks, I had realised this was likely a loop but hadn't bothered to look yet. I think the new commit should fix things.